compiler/passes/scopeResolve.cpp

```diff
@@ -1703,136 +1716,288 @@ static void    buildBreadthFirstModuleList(Vec<ModuleSymbol*>* modules,
+// scope
+// Lydia note (2015/06/26)
+// Semantic issue not handled by this function: when a parenthesis-less method
+// is defined by an outside module and the use of that module is at the same
+// scope as another symbol of the same name as the method, which symbol should
+// take precedent?  When the use is not present, both the prior version of
+// scopeResolve and the current version don't resolve the symbol, leaving the
+// decision to function resolution, which thinks it should have gotten the
+// method that wasn't available
+static bool methodMatched(BaseAST* scope, FnSymbol* method) {
+  if (method->_this->type->symbol == scope) {
+    return true;
+  } else {
+    BaseAST* curScope = getScope(scope);
+    // Traverse up the scopes either until we find this method or until there
+    // are no more scopes to traverse
+    while (curScope) {
+      if (TypeSymbol* ts = toTypeSymbol(scope)) {
+        // Are we in a type symbol?
+        if (Symbol* sym = getMethod(method->name, ts->type)) {
+          // Does that type symbol have a method with the same name?
+          if (sym == method) {
+            // Is it us?
+            return true;
+          } else {
+            // We are not in scope
+            return false;
+          }
+        }
+      }
+      curScope = getScope(curScope);
+    }
+    return false;
+  }
+}
 
-    if (symbolTable.count(scope) != 0) {
-      SymbolTableEntry* entry = symbolTable[scope];
+// inSymbolTable returns a Symbol* if there was an entry for this scope
+// that matched this name, NULL otherwise.
+static Symbol* inSymbolTable(BaseAST* scope, const char* name) {
+  if (symbolTable.count(scope) != 0) {
+    SymbolTableEntry* entry = symbolTable[scope];
+    if (entry->count(name) != 0) {
+      Symbol* sym = (*entry)[name];
+      // If the symbol found isn't a method, or it was a method and we are
+      // in the appropriate scope to add it (as determined by calling
+      // methodMatched), then return the symbol
+      if (!sym->hasFlag(FLAG_METHOD) || (methodMatched(scope, toFnSymbol(sym))))
+        return sym;
+    }
+  }
+  return NULL;
+}
 
-      if (entry->count(name) != 0) {
-        Symbol* sym = (*entry)[name];
-        symbols.set_add(sym);
+// If the current scope is an aggregate type, checks if the name refers to a
+// field or method on that type.  If a match is found, return it.  Otherwise
+// return NULL.
+static Symbol* inType(BaseAST* scope, const char* name) {
+  if (TypeSymbol* ts = toTypeSymbol(scope)) {
+    if (AggregateType* ct = toAggregateType(ts->type)) {
+      if (Symbol* sym = ct->getField(name, false)) {
+        return sym;
+      } else if (Symbol* fn = getMethod(name, ct)) {
+        // There is a method of that name, is it visible?
+        if (methodMatched(scope, toFnSymbol(fn))) {
+          return fn;
+        }
       }
     }
+  }
+  return NULL;
+}
+
+// Assumes that symbols contains nothing before entering this function
```

Comment:
It feels a little odd that the function is named "lookupThisScopeOnly" but then traverses through uses.